pkg/chart/chart.go

```diff
@@ -650,19 +668,23 @@ func (t *Testing) testRelease(namespace, release, releaseSelector string) error
 	return nil
 }
 
-func (t *Testing) generateInstallConfig(chart *Chart) (namespace, release, releaseSelector string, cleanup func()) {
+func (t *Testing) generateInstallConfig(chart *Chart) (namespace, release, releaseSelector string, cleanup func(*bool)) {
 	if t.config.Namespace != "" {
 		namespace = t.config.Namespace
 		release, _ = chart.CreateInstallParams(t.config.BuildId)
 		releaseSelector = fmt.Sprintf("%s=%s", t.config.ReleaseLabel, release)
-		cleanup = func() {
-			t.PrintEventsPodDetailsAndLogs(namespace, releaseSelector)
+		cleanup = func(failed *bool) {
+			if !t.config.LogFailedOnly || (*failed && t.config.LogFailedOnly) {
+				t.PrintEventsPodDetailsAndLogs(namespace, releaseSelector)
+			}
 			t.helm.DeleteRelease(namespace, release)
 		}
 	} else {
 		release, namespace = chart.CreateInstallParams(t.config.BuildId)
-		cleanup = func() {
-			t.PrintEventsPodDetailsAndLogs(namespace, releaseSelector)
+		cleanup = func(failed *bool) {
```

Comment:
Any reason why you use a pointer here?

Reply:
Since the [defer](https://github.com/deweya/chart-testing/blob/feat/hide-logs/pkg/chart/chart.go#L566) takes place before we know if the test [failed or not](https://github.com/deweya/chart-testing/blob/feat/hide-logs/pkg/chart/chart.go#L573-L581), I thought it would have to be passed by reference